packages/compiler/src/i18n/i18n_parser.ts

```diff
@@ -105,12 +105,13 @@ class _I18nVisitor implements html.Visitor {
   }
 
   visitAttribute(attribute: html.Attribute, context: I18nMessageVisitorContext): i18n.Node {
-    const node = this._visitTextWithInterpolation(attribute.value, attribute.sourceSpan, context);
+    const node = this._visitTextWithInterpolation(
+        attribute.value, attribute.valueSpan || attribute.sourceSpan, context, attribute.i18n);
     return context.visitNodeFn(attribute, node);
```

Comment:
The logic inside the `_visitTextWithInterpolation` function is a bit tricky. May be we can put it here instead?

```suggestion
    const node = this._visitTextWithInterpolation(
        attribute.value, attribute.valueSpan || attribute.sourceSpan, context);
    if (attribute.i18n) {
      // ... comments here ...
      reuseExistingSourceSpans(node, attribute.i18n.nodes[0]);
    }
    return context.visitNodeFn(attribute, node);
```
and with `visitText`:

```suggestion
  visitText(text: html.Text, context: I18nMessageVisitorContext): i18n.Node {
    const node = this._visitTextWithInterpolation(text.value, text.sourceSpan, context);
    if (text.i18n) {
      // ... comments here ...
      reuseExistingSourceSpans(node, text.i18n);
    }    
    return context.visitNodeFn(text, node);
  }
```
What do you think?

Reply:
Thanks for the review @AndrewKushnir. I agree that `_visitTextWithInterpolation()` is becoming too complex.
Rather than move the call outside of the method, though, I have refactored that method to be more maintainable.

Reply:
I did the refactoring in a separate commit, so that it is easier to see the actual fix for the issue.

Reply:
Thanks Pete! 👍